raimitigations/cohort/utils.py

```diff
@@ -0,0 +1,115 @@
+    (but not both). These parameters are the ones used in the constructor method
+    of the :class:`~raimitigations.cohort.CohortManager` class. Each metric is computed
+    separately for each set of predictions belonging to each of the existing cohorts.
+
+    :param X: the dataset containing the feature columns. This dataset is used to
+        filter the instances that belong to each cohort;
+    :param y_true: the true labels of the instances in the ``X`` dataset;
+    :param y_pred: the predicted labels;
+    :param cohort_def: a list of cohort definitions, a dictionary of cohort definitions,
+        or the path to a JSON file containing the definition of all cohorts. For more
+        details on this parameter, please check the :class:`~raimitigations.cohort.CohortManager`
+        class;
+    :param cohort_col: a list of column names or indices, from which one cohort is created for each
+        unique combination of values for these columns. This parameter is ignored if ``cohort_def``
+        is provided;
+    :param regression: if True, regression metrics are computed. If False, only classification
+        metrics are computed;
+    :param shared_th: if True, the binarization of the predictions is made using the same threshold
+        for all cohorts. The threshold used is the one computed for the entire dataset. If False,
+        a different threshold is computed for each cohort;
+    :return: a dataframe containing the metrics for the entire dataset and for all the defined
+        cohorts.
+    :rtype: pd.DataFrame
+    """
+
+    def _metric_tuple_to_dict(metric_dict, reg):
+        if reg:
+            metric_dict = {
+                "mse": metric_dict[MetricNames.MSE_KEY],
+                "rmse": metric_dict[MetricNames.RMSE_KEY],
+                "mae": metric_dict[MetricNames.MAE_KEY],
+                "r2": metric_dict[MetricNames.R2_KEY],
+            }
+        else:
+            metric_dict = {
+                "roc": metric_dict[MetricNames.AUC_KEY],
+                "precision": metric_dict[MetricNames.PREC_KEY],
+                "recall": metric_dict[MetricNames.RECALL_KEY],
+                "f1": metric_dict[MetricNames.F1_KEY],
+                "accuracy": metric_dict[MetricNames.ACC_KEY],
+                "threshold": metric_dict[MetricNames.TH],
+            }
+        return metric_dict
+
+    metrics_dict = {}
+    results = get_metrics(y_true, y_pred, regression, best_th_auc=True)
+    metrics_dict["all"] = _metric_tuple_to_dict(results, regression)
+    metrics_dict["all"]["cht_size"] = y_true.shape[0]
+
+    th = None
+    if shared_th:
+        th = metrics_dict["all"]["threshold"]
+
+    if cohort_def is None and cohort_col is None:
+        raise ValueError("ERROR: one of the two parameters must be provided: 'cohort_col' or 'cohort_def'.")
+    if cohort_def is None:
+        cht_manager = CohortManager(cohort_col=cohort_col)
+    else:
+        cht_manager = CohortManager(cohort_def=cohort_def)
```

Comment:
Sometimes you write "cohort_*", sometimes "cht_*". I think we should be consistent. I prefer "cohort_*".